src/com/amazon/ionschema/internal/constraint/ValidValues.kt

```diff
@@ -80,8 +88,21 @@ internal class ValidValues(
             }
         } else {
             val v = value.withoutTypeAnnotations()
-            if (!validValues!!.contains(v)) {
-                issues.add(Violation(ion, "invalid_value", "invalid value $v"))
+            // check any range exists in valid_values list and validate
+            if (anyRangeValues) {
+                if (!validValues!!.any {possibility ->
+                            if (isValidRange(possibility)) {
+                                buildRange(possibility)!!.contains(v)
```

Comment:
Every time `validate` is called, this is going to:
* Iterate over everything in `valid_values` to see if there are any ranges with `anyRangeValues` calling `isValidRange`.
* Iterate over everything in `valid_values` to see if there are any ranges with `isValidRange`; for each one that is, it will:
   * Build the range with `buildRange`, which in turn checks to see if it's a valid range with `isValidRange`

Two performance observations:
1. There's a fair amount of redundant evaluation happening here. I think you can eliminate a lot of it by just calling `buildRange` on every `possibility` and then filtering out the `null`s that it produces. If a range makes it through, then you can call `contains()` on it.
2. The Ion representing the range is never going to change, so the range it builds will be the same every time. It seems like something we could cache somewhere.

We can punt on #2 as I imagine that's happening in all of the validators right now, but we should address #1 in this PR.